Controller/Checkout/Cart.php

```diff
@@ -44,12 +46,24 @@ public function execute()
 
         unset($params['quote_id']);
 
-        try {
-            $quoteIdMask = $this->quoteIdMaskFactory->create()->load($quoteId, 'masked_id');
-            $quote = $this->quoteRepository->get($quoteIdMask->getQuoteId());
-            $this->cart->setQuote($quote);
-            $this->cart->save();
-        } catch (\Magento\Framework\Exception\NoSuchEntityException $ex) {
+        // Check if the quote_id has kx_identifier, if yes, retreive active quote for customer, if not get QuoteId from masked QuoteId
+        if (strpos($quoteId, "kx_identifier_") !== false){
+            $customerId = base64_decode( str_replace("kx_identifier_", "", $quoteId) );
+            try {
+                $quote = $this->quoteRepository->getActiveForCustomer($customerId);
```

Comment:
is it possible for a customer to have more than one active quote at a time? like if they opened 2 different sessions one on their phone and one on their laptop?

Reply:
I answered it in the above comment